zone/raids.cpp

```diff
@@ -2203,3 +2257,734 @@ void Raid::SetNewRaidLeader(uint32 i)
+		XTargetType assist_type;
+		XTargetType assist_target_type;
+		int32       slot;
+	};
+
+	std::vector<AssistUpdate> assist_updates = {
+		AssistUpdate{.assist_type = RaidAssist1, .assist_target_type = RaidAssist1Target, .slot = MAIN_ASSIST_1_SLOT},
+		AssistUpdate{.assist_type = RaidAssist2, .assist_target_type = RaidAssist2Target, .slot = MAIN_ASSIST_2_SLOT},
+		AssistUpdate{.assist_type = RaidAssist3, .assist_target_type = RaidAssist3Target, .slot = MAIN_ASSIST_3_SLOT},
+	};
+
+	for (const auto& u : assist_updates) {
+		if (strlen(main_assister_pcs[u.slot]) > 0) {
+			auto m = entity_list.GetMob(main_assister_pcs[u.slot]);
+			if (m) {
+				UpdateXTargetType(u.assist_type, m, m->GetName());
+				auto n = m->GetTarget();
+				if (n && n->GetHP() > 0) {
+					UpdateXTargetType(u.assist_target_type, n, n->GetName());
+				}
+				else {
+					UpdateXTargetType(u.assist_target_type, nullptr);
+				}
+			}
+		}
+		else {
+			UpdateXTargetType(u.assist_type, nullptr);
+			UpdateXTargetType(u.assist_target_type, nullptr);
+		}
+	}
+
+	struct MarkedUpdate {
+		XTargetType mark_target;
+		int32       slot;
+	};
+
+	std::vector<MarkedUpdate> marked_updates = {
+		MarkedUpdate{.mark_target = RaidMarkTarget1, .slot = MAIN_MARKER_1_SLOT},
+		MarkedUpdate{.mark_target = RaidMarkTarget2, .slot = MAIN_MARKER_2_SLOT},
+		MarkedUpdate{.mark_target = RaidMarkTarget3, .slot = MAIN_MARKER_3_SLOT},
+	};
+
+	for (auto& u : marked_updates) {
+		if (marked_npcs[u.slot]) {
+			auto m = entity_list.GetMob(marked_npcs[u.slot]);
+			if (m && m->GetHP() > 0) {
+				UpdateXTargetType(u.mark_target, m, m->GetName());
+			}
+			else {
+				UpdateXTargetType(u.mark_target, nullptr);
+			}
+		}
+		else {
+			UpdateXTargetType(u.mark_target, nullptr);
+		}
+	}
+}
+
+void Raid::DelegateAbilityMark(Mob* delegator, const char* delegatee)
+{
+	auto raid_delegatee = entity_list.GetRaidByName(delegatee);
+	if (!raid_delegatee) {
+		delegator->CastToClient()->MessageString(Chat::Cyan, 5082, delegatee);
+		return;
+	}
+	uint32 raid_delegatee_id = raid_delegatee->GetID();
+	uint32 raid_delegator_id = GetID();
+	if (raid_delegatee_id != raid_delegator_id) {
+		delegator->CastToClient()->MessageString(Chat::Cyan, NOT_IN_YOUR_RAID, delegatee);
+		return;
+	}
+
+	auto rm = &members[GetPlayerIndex(delegatee)];
+	if (!rm) {
+		return;
+	}
+	auto c = rm->member;
+	if (!c) {
+		return;
+	}
+
+	auto slot = FindNextRaidDelegateSlot(FindNextMarkerSlot);
+	auto mm = rm->main_marker;
+
+	if (slot == -1 && !mm) {
+		delegator->CastToClient()->MessageString(Chat::Cyan, MAX_MAIN_RAID_MARKERS);
+		return;
+	}
+
+	auto outapp = new EQApplicationPacket(OP_RaidDelegateAbility, sizeof(DelegateAbility_Struct));
+	DelegateAbility_Struct* das = (DelegateAbility_Struct*)outapp->pBuffer;
+	if (mm) {
+		das->Action = ClearDelegate;
+		memset(main_marker_pcs[mm - 1], 0, 64);
+		rm->main_marker = 0;
+		std::string query = StringFormat("UPDATE raid_members SET is_marker = 0 WHERE raidid = %i AND name= '%s';",
```

Comment:
Can do the same thing for this as with the assister method.
```cpp
static int UpdateRaidMarker(
	Database& db,
	int32_t raid_id,
	const std::string& character_name,
	uint8_t marker
) {
	auto results = db.QueryDatabase(
		fmt::format(
			"UPDATE {} SET is_marker = {} WHERE raidid = {} AND `name` = '{}'",
			TableName(),
			marker,
			raid_id,
			Strings::Escape(character_name)
		)
	);

	return results.Success() ? results.RowsAffected() : 0;
}
```

```cpp
int updated = RaidMembersRepository::UpdateRaidAssister(database, GetID(), delegatee, 0);
if (!updated) {
    LogError("Unable to clear raid main marker for player [{}]", delegatee);
}
```

Reply:
Updated all db routines.